Standard library fails to build for arm64_32-apple-watchos

Building the latest nightly with `--target arm64_32-apple-watchos`, fails with the following:

```rust
  error[E0308]: mismatched types
     --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/sync/condvar.rs:123:21
      |
  123 |             tv_sec: dur.as_secs() as i64,
      |                     ^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `i64`
  
  error[E0308]: mismatched types
     --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/sync/condvar.rs:124:22
      |
  124 |             tv_nsec: dur.subsec_nanos() as i64,
      |                      ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `i64`
```

### Meta

`rustc --version --verbose`:
```
rustc 1.92.0-nightly (57ef8d642 2025-10-15)
binary: rustc
commit-hash: 57ef8d642d21965304bde849bab4f389b4353e27
commit-date: 2025-10-15
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.3
```
<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

--- a/library/std/src/sys/pal/unix/sync/condvar.rs
+++ b/library/std/src/sys/pal/unix/sync/condvar.rs
@@ -1,6 +1,10 @@
 use super::Mutex;
 use crate::cell::UnsafeCell;
 use crate::pin::Pin;
+#[cfg(not(target_os = "nto"))]
+use crate::sys::pal::time::TIMESPEC_MAX;
+#[cfg(target_os = "nto")]
+use crate::sys::pal::time::TIMESPEC_MAX_CAPPED;
 use crate::time::Duration;
 
 pub struct Condvar {
@@ -51,10 +55,6 @@ impl Condvar {
     /// * `mutex` must be locked by the current thread.
     /// * This condition variable may only be used with the same mutex.
     pub unsafe fn wait_timeout(&self, mutex: Pin<&Mutex>, dur: Duration) -> bool {
-        #[cfg(not(target_os = "nto"))]
-        use crate::sys::pal::time::TIMESPEC_MAX;
-        #[cfg(target_os = "nto")]
-        use crate::sys::pal::time::TIMESPEC_MAX_CAPPED;
         use crate::sys::pal::time::Timespec;
 
         let mutex = mutex.raw();
@@ -118,10 +118,12 @@ impl Condvar {
 
         let (dur, clamped) = if dur <= MAX_DURATION { (dur, false) } else { (MAX_DURATION, true) };
 
-        let timeout = libc::timespec {
-            // This cannot overflow because of the clamping above.
-            tv_sec: dur.as_secs() as i64,
-            tv_nsec: dur.subsec_nanos() as i64,
+        // This can overflow on 32-bit platforms, but not on 64-bit because of the clamping above.
+        let timeout = if let Ok(tv_sec) = dur.as_secs().try_into() {
+            libc::timespec { tv_sec, tv_nsec: dur.subsec_nanos() as _ }
+        } else {
+            // This is less than `MAX_DURATION` on 32-bit platforms.
+            TIMESPEC_MAX
         };
 
         let r = unsafe { libc::pthread_cond_timedwait_relative_np(self.raw(), mutex, &timeout) };

